packages/state-transition/src/cache/epochCache.ts

```diff
@@ -383,16 +422,43 @@ export class EpochCache {
       throw Error("totalActiveBalanceIncrements >= Number.MAX_SAFE_INTEGER. MAX_EFFECTIVE_BALANCE is too low.");
     }
 
-    const currentShuffling =
-      cachedCurrentShuffling ??
-      computeEpochShuffling(state, currentActiveIndices, currentActiveIndices.length, currentEpoch);
-    const previousShuffling =
-      cachedPreviousShuffling ??
-      (isGenesis
-        ? currentShuffling
-        : computeEpochShuffling(state, previousActiveIndices, previousActiveIndices.length, previousEpoch));
-    const nextShuffling =
-      cachedNextShuffling ?? computeEpochShuffling(state, nextActiveIndices, nextActiveIndices.length, nextEpoch);
+    const previousActiveIndices = new Uint32Array(previousActiveIndicesAsNumberArray);
```

Comment:
only create these `Uint32Array` objects on demand in the below if/else clause

Reply:
Ok.  I added a scope around the if/else and constrained the two that dont get retuned.